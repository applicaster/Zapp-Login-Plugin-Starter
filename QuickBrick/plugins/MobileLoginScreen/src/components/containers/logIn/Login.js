/* eslint-disable react-native/no-inline-styles */
import React, {Component} from 'react'
import propTypes from 'prop-types'
import {
  View,
  Text,
  KeyboardAvoidingView,
  ScrollView,
  StyleSheet,
  Image,
} from 'react-native'
import {pathOr} from 'ramda'
import colors from '../../../style/color'

// Components
import InputField from '../../commons/inputField/InputField'
import RoundedButton from '../../commons/roundedButton/RoundedButton'

// Action
import {loginRequest} from '../../../actions/authAPI'

// Helpers
import {validateEmail, validateNotEmpty} from '../../../helper/validations'

export default class Login extends Component {
  constructor(props) {
    super(props)
    this.state = {
      email: '',
      password: '',
    }
  }

  static propTypes = {
    screenData: propTypes.object,
    callback: propTypes.func,
    payload: propTypes.object,
  }

  updateForm = (value, label) => {
    this.setState({
      [label]: value,
    })
  }

  isLoginButtonEnable = () => {
    const {email, password} = this.state
    return validateEmail(email) === '' && validateNotEmpty(password) === ''
  }

  requestLoginAPI = () => {
    const {callback, payload, screenData} = this.props
    const isTesting = pathOr(false, ['general', 'is_testing'], screenData)
    loginRequest('/TEST_URL', this.state, isTesting) // THIS IS IN TEST MODE !!!!
      .then(res => {
        callback({
          success: true,
          payload,
        })
      })
    /***
     ***** ADD YOUR CUSTOM LOGIC HERE !!!! *****
    ex: save on localStorage token received from API
    ***/
  }

  render() {
    const {screenData} = this.props
    const isEnabled = this.isLoginButtonEnable()

    // Customizable layoout
    const appColorBackground = pathOr(
      colors.darkblue,
      ['general', 'app_bgcolor'],
      screenData,
    )
    const appFontColor = pathOr(
      colors.black,
      ['general', 'font_color'],
      screenData,
    )

    return (
      <KeyboardAvoidingView
        style={{...styles.wrapper, backgroundColor: appColorBackground}}
        behavior="padding">
        <View style={styles.scrollViewWrapper}>
          <ScrollView style={styles.avoidView}>
            <Image
              style={styles.logo}
              source={require('../../../style/images/applicaster_logo.png')}
            />
            <Text style={{...styles.loginHeader, color: appFontColor}}>
              Screen Login Plugin
            </Text>
            <InputField
              labelText="EMAIL ADDRESS"
              labelTextSize={14}
              labelColor={colors.white}
              textColor={colors.white}
              borderBottomColor={colors.white}
              inputType="email"
              customStyle={{marginBottom: 30}}
              inputValidation={validateEmail}
              onUpdateValue={this.updateForm}
            />
            <InputField
              labelText="PASSWORD"
              labelTextSize={14}
              labelColor={colors.white}
              textColor={colors.white}
              borderBottomColor={colors.white}
              inputType="password"
              customStyle={{marginBottom: 30}}
              inputValidation={validateNotEmpty}
              onUpdateValue={this.updateForm}
            />
            <RoundedButton
              text="Login"
              isEnabled={isEnabled}
              action={this.requestLoginAPI}
            />
          </ScrollView>
        </View>
      </KeyboardAvoidingView>
    )
  }
}

const styles = StyleSheet.create({
  wrapper: {
    display: 'flex',
    flex: 1,
  },
  scrollViewWrapper: {
    marginTop: 20,
    flex: 1,
  },
  avoidView: {
    paddingLeft: 30,
    paddingRight: 30,
    paddingTop: 20,
    flex: 1,
  },
  logo: {
    width: '100%',
    height: 50,
    marginTop: 10,
    marginBottom: 40,
  },
  loginHeader: {
    fontSize: 28,
    fontWeight: '300',
    marginBottom: 40,
  },
})
