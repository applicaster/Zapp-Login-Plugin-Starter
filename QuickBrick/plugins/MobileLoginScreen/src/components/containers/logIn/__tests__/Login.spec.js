import React from 'react'
import renderer from 'react-test-renderer'

import Login from '../Login'

describe('@Login', () => {
  describe('Render', () => {
    it('Render with default props', () => {
      const root = renderer.create(<Login />)
      expect(root.toJSON()).toMatchSnapshot()
    })
  })
})
