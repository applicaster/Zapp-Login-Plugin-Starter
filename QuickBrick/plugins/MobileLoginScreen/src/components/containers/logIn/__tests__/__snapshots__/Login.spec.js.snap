// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@Login Render Render with default props 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#2132B3",
        "display": "flex",
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginTop": 20,
      }
    }
  >
    <RCTScrollView
      style={
        Object {
          "flex": 1,
          "paddingLeft": 30,
          "paddingRight": 30,
          "paddingTop": 20,
        }
      }
    >
      <View>
        <Image
          source={
            Object {
              "testUri": "../../../plugins/MobileLoginScreen/src/style/images/applicaster_logo.png",
            }
          }
          style={
            Object {
              "height": 50,
              "marginBottom": 40,
              "marginTop": 10,
              "width": "100%",
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#000000",
              "fontSize": 28,
              "fontWeight": "300",
              "marginBottom": 40,
            }
          }
        >
          Screen Login Plugin
        </Text>
        <View
          style={
            Array [
              Object {
                "marginBottom": 30,
              },
              Object {
                "display": "flex",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontSize": 14,
                },
                Object {
                  "fontWeight": "700",
                  "marginBottom": 10,
                },
              ]
            }
          >
            EMAIL ADDRESS
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            rejectResponderTermination={true}
            secureTextEntry={false}
            style={
              Array [
                Object {
                  "borderBottomColor": "#ffffff",
                  "color": "#ffffff",
                },
                Object {
                  "borderBottomWidth": 1,
                  "paddingBottom": 5,
                  "paddingTop": 5,
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Array [
              Object {
                "marginBottom": 30,
              },
              Object {
                "display": "flex",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontSize": 14,
                },
                Object {
                  "fontWeight": "700",
                  "marginBottom": 10,
                },
              ]
            }
          >
            PASSWORD
          </Text>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
                "position": "absolute",
                "right": 0,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontWeight": "700",
                }
              }
            >
              Show
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Array [
                Object {
                  "borderBottomColor": "#ffffff",
                  "color": "#ffffff",
                },
                Object {
                  "borderBottomWidth": 1,
                  "paddingBottom": 5,
                  "paddingTop": 5,
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderColor": "#000000",
                "borderRadius": 40,
                "borderWidth": 1,
                "display": "flex",
                "padding": 15,
              },
              Object {
                "backgroundColor": "#CCCCCC",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "textAlign": "center",
                "width": "100%",
              }
            }
          >
            Login
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
